src/XrdHttp/XrdHttpReq.cc

```diff
@@ -661,6 +661,13 @@ bool XrdHttpReq::Redir(XrdXrootd::Bridge::Context &info, //!< the result context
   else
     redirdest = "Location: http://";
   
+  // port -1 signals redirect to local filesystem
+  // switch to file:// protocol
+  if (port == -1)
```

Comment:
Ummm, I think the comment is wrong here. When port is < 0 then the target is a full URL not a host?token:port target. I presume you do set port to -1 and the target is a full URL with "file:://" as the protocol. But it might also be a regular full URL. The code here should take that into account and not merely assume it's a local file redirect. Technically, the check for '-1' is correct because it is documented as such. However, we are considering loosing it and allowing for any value less than zero to accommodate future needs. So, I'd suggest checking it's merely negative so we don't have to revisit this when it changes.

Reply:
Hi Andy, thanks.

do you merely want a change such as this:
```
  // port < 0 signals switch to full URL
  if (port < 0)
  {
    TRACE(REQ, " XrdHttpReq::Redir Switching to file:// ");
    redirdest = "Location: file://";
  }
```

And I want to make sure I understand you correctly:
Wouldn't checking for port < 0 make everything more ambiguous?
Because new use cases that would utilize other ports < 0, would trigger the local file redirect as well?

Or do you mean that there will only be the one use case for ports < 0, and that will remain to be local file redirect?
And I am unclear on the term "regular full URL". In what other cases does xrootd handle non host?token:port targets than in the file://-protocol?

Cheers,
Paul

Reply:
Hi Paul,

The new definition is that if the port number is < 0 then the redirect target is a full URL. In your case, you need to check if the URL starts with "file://" and effect the redirection to the local file. Otherwise, the redirection should proceed as a normal full URL redirect for some other protocol. So, do not assume that just because you got a full URL it is for local file redirection.

Reply:
Hi Andy,

ok I see what you mean.
I hope the newest commit will be sufficient.
I previously did not return the file://-prefix from the plug-in, just the path.

In the root-protocol, a port of -1 was sufficient.
I can get started on changing this too, checking for the presence of "file://" there as well.

Cheers,
Paul